### PR TITLE
Exercise 11

### PR DESCRIPTION
**Pull request by pat-morris**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,28 +3,25 @@ services:
...
-  - docker tag $DOCKER_HUB_USERNAME/todo-app:latest registry.heroku.com/$HEROKU_APP/web
 
 script:
   - docker run test tests
   - docker run test tests_integration
   - >
     docker run
-    -e MONGO_DB_USERNAME
-    -e MONGO_DB_PASSWORD
+    -e MONGO_DB_PRIMARY_CONNECTION_STRING
```
Comment (JackMeadDev) [inline .travis.yml:16]: The error thrown by travis suggests that this environment variable is being passed through empty - in that case pymongo tries to connect to a locally running mongo instance and the error message mentions `localhost:27017`
To track this down, I'd start by checking the env variable was encrypted correctly - adding the variable via travis' gui & removing it from your config for example may make that quicker/easier to verify

File: .travis.yml [hunk 2]
```diff
@@ -3,28 +3,25 @@ services:
...
   - docker push $DOCKER_HUB_USERNAME/todo-app:latest
 
 deploy:
   provider: script
-  script: bash deploy.sh
+  script: 
+      curl -dH -X POST "$AZURE_WEBHOOK"
   on:
-    branch: master
+    branch: exercise-11
```
Comment (JackMeadDev) [inline .travis.yml]: Don't forget to switch this back to master on merge

**Review by JackMeadDev — APPROVED** [review]
This looks good, I can see that your app is hosted in Azure and working. 

I've left a couple of thoughts below, and we should add some documentation to mention in our Readme that the app is automatically deployed on master and the URL it's available at, but happy for you to do that in M12 as it will likely change there anyway. I would also recommend investigating the error in your end-to-end test, the error message suggests that the environment variable isn't being correctly passed through, and verifying the application deployment is still succeeding in our pipeline is important

File: .travis.yml (hunk 1 above)
Comment (pat-morris) [reply .travis.yml:16]: @JackMead - Bingo! Thank you, this has been driving me nuts. The build is now succeeding and the E2E test completed successfully on Travis.

**Pull request merged**